src/bit-systems/media-loading.ts

```diff
@@ -216,21 +221,20 @@ function* loadAndAnimateMedia(world: HubsWorld, eid: EntityID, clearRollbacks: C
   if (MediaLoader.flags[eid] & MEDIA_LOADER_FLAGS.ANIMATE_LOAD) {
     yield* animateScale(world, media);
   }
-  removeComponent(world, MediaLoader, eid);
 
-  if (media) {
-    if (hasComponent(world, MediaLoaded, media)) {
-      const box = getBox(world, eid, media);
-      addComponent(world, MediaContentBounds, eid);
-      box.getSize(tmpVector);
-      MediaContentBounds.bounds[eid].set(tmpVector.toArray());
-    }
-    // TODO update scale?
-    inflatePhysicsShape(world, eid, {
-      type: hasComponent(world, GLTFModel, media) ? Shape.HULL : Shape.BOX,
-      minHalfExtent: 0.04
-    });
-  }
+  const box = getBox(world, eid, media);
+  addComponent(world, MediaContentBounds, eid);
+  box.getSize(tmpVector);
+  MediaContentBounds.bounds[eid].set(tmpVector.toArray());
+
+  // TODO update scale?
+  inflatePhysicsShape(world, media, {
+    type: hasComponent(world, GLTFModel, media) ? Shape.HULL : Shape.BOX,
+    minHalfExtent: 0.04
+  });
+
+  addComponent(world, MediaLoaded, media);
+  removeComponent(world, MediaLoader, eid);
```

Comment:
Just in case, this changes the timing of when `MediaLoaded` is added to an entity. It was added right after when media is loaded (eg: before `animateScale()` starts) while with this change it is added everything is finished (eg: after `animateScale()` ends). Did you confirm this timing change won't affect any other system? (Sorry I didn't have time to check the other codes by myself.)

Reply:
This affects when media playback starts: https://github.com/mozilla/hubs/blob/95a4deefafc548ab0104b6af32bf5ddb77d96454/src/bit-systems/video-system.ts#L49 but my understanding is that this is a more desirable behavior as we probably want a video playback to start when it's fully visible rather than when it's still scaling.
It also affects when a media-frame can capture an entity https://github.com/mozilla/hubs/blob/95a4deefafc548ab0104b6af32bf5ddb77d96454/src/systems/bit-media-frames.js#L114 and I also think that it makes sense that works this way unless I'm missing something.

Reply:
I've added a commit to don't recenter/resize for objects without volume, I think that makes more sense.

Reply:
> I've added a commit to don't recenter/resize for objects without volume, I think that makes more sense.

I agree with that it makes more sense.